src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -760,6 +776,20 @@ namespace AppInstaller::Logging
         AICLI_LOG(CLI, Error, << type << " repair failed: " << errorCode);
     }
 
+    Settings::ExperimentState TelemetryTraceLogger::GetExperimentState(Experiment::Key key)
+    {
+#ifndef AICLI_DISABLE_TEST_HOOKS
+        m_summary.Experiments.clear();
+#endif
+
+        if (m_summary.Experiments.find(key) == m_summary.Experiments.end())
+        {
+            m_summary.Experiments[key] =  Experiment::GetStateInternal(key);

```

Comment:
Do we need to cache this? Doesn't seem like we use it anywhere but the summary event, which fires only once AFAIK

Reply:
Getting the state of an experiment is used when applying that experiment in a specific use case and again that state will be reported in the summary event. This cache will also only contain the experiments that has been used during a workflow, and only those will be reported in the summary event.